librosa/util/utils.py

```diff
@@ -627,8 +636,26 @@ def normalize(S, norm=np.inf, axis=0):
     else:
         raise ParameterError('Unsupported norm: {}'.format(repr(norm)))
 
+    # get the vectors-ids smaller than the threshold
+    small_values_idx = (length < threshold)
+
     # Avoid div-by-zero
-    length[length < SMALL_FLOAT] = 1.0
+    length[small_values_idx] = 1.0
+
+    if mode and np.sum(small_values_idx):
+        # init empty slice object
+        slices = [slice(None)] * S.ndim
+
+        # set slice for target-axis
+        target_axis_idx = int(np.logical_not(axis))
```

Comment:
> Any idea of how it should behave in higher dimensions?

The short answer is that the entire slice that you'd be normalizing should be set.

For instance, if you have a 3d array, and you're normalizing along `axis=1`, then 
``S[:, i, :]`` should be normalized all as one "vector".  If this sounds strange, imagine that you first permuted the 0 and 1 axes, so that you have ``S_permute[i, :, :]``, and then flatten the last two axes so that you just have ``S_permute_flat[i, :]``.  Any normalization you do to ``S`` should be equivalent to the corresponding vector-wise normalization on the flattened version.

In terms of what this means for your code, I still don't fully understand the implementation here.

What I would expect is that for each index ``i`` where the length is below the threshold, you do  something like
```python
index = [slice(None)] * S.ndim

i = too_short_indices

for i in too_short_indices:
    index[axis] = i
    S[tuple(index)] = #filler
```
Maybe this assignment can be vectorized so you don't need the loop?  At any rate, the logical_not seems strange, and will not generalize to high-dimensions.